lib/livebook/hubs.ex

```diff
@@ -82,22 +82,31 @@ defmodule Livebook.Hubs do
     attributes = struct |> Map.from_struct() |> Map.to_list()
     :ok = Storage.insert(@namespace, struct.id, attributes)
     :ok = connect_hub(struct)
-    :ok = Broadcasts.hubs_metadata_changed()
+    :ok = Broadcasts.hub_changed()
 
     struct
   end
 
   @doc false
   def delete_hub(id) do
     with {:ok, hub} <- get_hub(id) do
-      :ok = Provider.disconnect(hub)
+      :ok = Broadcasts.hub_changed()
       :ok = Storage.delete(@namespace, id)
-      :ok = Broadcasts.hubs_metadata_changed()
+      :ok = disconnect_hub(hub)
     end
 
     :ok
   end
 
+  defp disconnect_hub(hub) do
+    Task.async(fn ->
```

Comment:
We will need a task supervisor. Otherwise the current process can terminate and then the hub is never disconnected.

Reply:
We would use Task.Supervisor.start_child in that case. Notice that every time you use `async`, you must use `await`.